src/VisionModel/resnet.jl

```diff
@@ -0,0 +1,82 @@
+using Flux
+
+basicblock(inplanes::Int, outchannels, downsample::Bool = false) = downsample ? 
+  Chain(Conv((3, 3), inplanes => outchannels[1], stride = 2),
+        BatchNorm(inplanes, λ = relu),
+        Conv((3, 3), inplanes => outchannels[2], stride = 1, pad = 1),
+        BatchNorm(outchannels, λ=relu)) : 
+  Chain(Conv((3, 3), inplanes => outchannels[1], stride = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((3, 3), inplanes => outchannels[2], stride = 1, pad = 1),
+        BatchNorm(outchannels, λ=relu))
+
+bottleneck(inplanes::Int, outchannels, downsample::Bool = false) = downsample ?
+  Chain(Conv((1, 1), inplanes => outchannels[1], stride = 2),
+        BatchNorm(inplanes, λ = relu),
+        Conv((3, 3), inplanes => outchannels[2], stride = 1, pad = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((1, 1), inplanes => outchannels[3], stride = 1),
+        BatchNorm(outchannels, λ = relu)) :
+  Chain(Conv((1, 1), inplanes => outchannels[1], stride = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((3, 3), inplanes => outchannels[2], stride = 1, pad = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((1, 1), inplanes => outchannels[3], stride = 1),
+        BatchNorm(outchannels, λ = relu))
+
+projection(inplanes::Int, outplanes::Int, stride::Int) = Chain(Conv((1, 1), inplanes => outplanes, stride=stride),
+                                                               BatchNorm((outplanes), λ=relu))
+
+identity(inplanes::Int, outplanes::Int, stride::Int) = +
```

Comment:
This will need to be modified to perform zero-padding when `inplanes < outplanes`.